src/neo/SmartContract/Native/Tokens/GasToken.cs

```diff
@@ -27,6 +27,7 @@ protected override void OnPersist(ApplicationEngine engine)
             long totalNetworkFee = 0;
             foreach (Transaction tx in engine.Snapshot.PersistingBlock.Transactions)
             {
+                NEO.DistributeGas(engine, tx.Sender);
```

Comment:
@shargon, maybe it should only distributeGAS for the `tx.Sender` that needs it.
In this sense, perhaps we could have a flag before `return balance + unclaimed >= fee;`

Reply:
Please, take a look again